src/App.js

```diff
@@ -67,39 +71,137 @@ const App = () => {
+      const options = {
+        method: 'DELETE',
+        headers: {
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+          'Content-Type': 'application/json'
+        },
+      };
+
+      const response = await fetch(url, options);
+
+      if(response.ok) {
+        console.log('The todo has been successfully deleted.');
+      } else if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+
+      const dataResponse = await response.json();
+      console.log(dataResponse);
+
+      fetchData();
+      
+    } catch (error) {
+      console.log(error.message);
+      return null;
+    }
+  }
+
+  const patchTodo = async (todo) => {
+    try {
+      const id = todo.id;
+      const dataToAirtable = {
+        fields: {
+          title: todo.title //to be updated
+        }
+      }
+      console.log(dataToAirtable);
+      const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}/${id}`;
+      const options = {
+        method: 'PATCH',
+        headers: {
+          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(dataToAirtable),
+      };
+
+      const response = await fetch(url, options);
+
+      console.log(response);
+      if(response.ok) {
+        console.log('The todo has been successfully updated.');
+      } else if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
 
-  React.useEffect(() => {
-    let stringifiedTodoList = JSON.stringify(todoList);
-    if (isLoading === false) {
-      localStorage.setItem('savedTodoList', stringifiedTodoList);
+      const dataResponse = await response.json();
+      console.log(dataResponse);
+      setTodoList([...todoList, dataResponse.fields]);
+
+      fetchData();
+
+    } catch (error) {
+      console.log(error.message);
+      return null;
     }
-  }, [todoList]);
+  }
+
+  React.useEffect(() => {fetchData();},[]);
 
+  // The use effect below keeps giving me this message in the terminal/console:
+  // "React Hook React.useEffect has a missing dependency: 'isLoading'. Either include it or remove the dependency array"
+  // I commented it out and the app worked fine, so I'm thinking of removing it altogether, unless I'm adviced against it.
+  // React.useEffect(() => {
+  //   let stringifiedTodoList = JSON.stringify(todoList);
+  //   if (isLoading === false) {
+  //     localStorage.setItem('savedTodoList', stringifiedTodoList);
+  //   }
+  // }, [todoList]);
+
+  // Since fetchData() is called in postTodo(), and setTodoList updates the state with the fetched airtable data,
+  // I might delete the last 2 lines of this addTodo function, (or delete this function altogether and
+  // pass postTodo() straight to the onAddTodo={} handler)
   const addTodo = (newTodo) => {
+
     setTodoList([...todoList, newTodo]);
```

Comment:
I appreciate your comments, and they certainly make sense to me.
 I see you're still working on POST and PATCH. When I tried to add a new task, I got a warning. 
 
<img width="856" alt="Screenshot 2024-02-05 at 2 57 58 PM" src="https://github.com/mariasantiago01/react-todo/assets/11164899/f3418f20-141c-41ae-b914-dcbb67ae73b2">

You may need to remove this line 180 `setTodoList([...todoList, newTodo])`.
And update the state on line 82 this way `setTodoList([...todoList, dataResponse]);`
Unfortunately, I couldn't PATCH successfully.